src/pkg/build/build.go

```diff
@@ -0,0 +1,298 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package build
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"syscall"
+
+	"github.com/singularityware/singularity/src/pkg/build/assemblers"
+	"github.com/singularityware/singularity/src/pkg/build/sources"
+	"github.com/singularityware/singularity/src/pkg/build/types"
+	"github.com/singularityware/singularity/src/pkg/buildcfg"
+	"github.com/singularityware/singularity/src/pkg/sylog"
+	syexec "github.com/singularityware/singularity/src/pkg/util/exec"
+	"github.com/singularityware/singularity/src/runtime/engines/common/config"
+	"github.com/singularityware/singularity/src/runtime/engines/common/oci"
+	"github.com/singularityware/singularity/src/runtime/engines/imgbuild"
+)
+
+// Build is an abstracted way to look at the entire build process.
+// For example calling NewBuild() will return this object.
+// From there we can call Full() on this build object, which will:
+// 		Call Bundle() to obtain all data needed to execute the specified build locally on the machine
+// 		Execute all of a definition using AllSections()
+// 		And finally call Assemble() to create our container image
+type Build struct {
+	dest        string           // Location for container after build is complete
```

Comment:
Thanks for adding the comments. Can you change the formatting a bit? To work with `godoc` you should change:

`dest        string           // Location for container after build is complete`
to
```
// dest is the location for container after build is complete
dest string
```

Reply:
Done